core/tx_pool.go

```diff
@@ -1262,12 +1257,13 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	}
 	pool.currentState = statedb
 	pool.pendingNonces = newTxNoncer(statedb)
-	pool.currentMaxGas = CalcGasLimit(pool.chain.CurrentBlock(), statedb)
 
+	pool.currentVMRunner = pool.chain.NewEVMRunner(newHead, statedb)
```

Comment:
I notice this uses `newHead` while the `currentMaxGas` before used `CurrentBlock()`.  I guess they should be the same, but if in some crazy race condition they were somehow to differ then it seems like `newHead` would be more correct.

Reply:
true, i'd say new code fixes the race condition. 

And fixing it doesn't generate a HF, so we're fine